shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.mm

```diff
@@ -41,18 +91,21 @@ - (void)blockUntilFrameAvailable {
 
   _beginResizeWaiting = YES;
 
-  while (CGSizeEqualToSize(_contentSize, CGSizeZero) && !_shuttingDown) {
+  while (![self someViewsHaveFrame] && !_shuttingDown) {
     _condBlockBeginResize.wait(lock);
     [self drain];
   }
 
   _beginResizeWaiting = NO;
 }
 
-- (void)beginResize:(CGSize)size notify:(nonnull dispatch_block_t)notify {
+- (void)beginResizeForView:(int64_t)viewId
+                      size:(CGSize)size
+                    notify:(nonnull dispatch_block_t)notify {
+  dispatch_assert_queue(_mainQueue);
   std::unique_lock<std::mutex> lock(_mutex);
```

Comment:
I haven't tested this yet, but does the assert actually pass? I don't think the event handler actually runs from main dispatch queue (meaning while `CFRunLoop` is draining main dispatch queue), so technically this should fail. But then again, main dispatch queue is somewhat special so maybe the assert also checks for main thread.

Reply:
Yes, it actually passes. I've merged this patch into our multiview-prototype branch and ran it with a multiwindow MVP app (see https://github.com/goderbauer/mvp) and the windows can correctly be resized without crashing.

I've also once incorrectly called `registerView:` in the test main instead of in `dispatchMainTask` and the test crashed as expected.

To explain, macOS associates the main queue with the main thread, and all other queues with a dynamic thread in a thread pool, as [the documentation](https://developer.apple.com/documentation/dispatch/dispatch_queue) says,

> Except for the dispatch queue representing your app's main thread, the system makes no guarantees about which thread it uses to execute a task.